consensus/core/src/leader_schedule.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use rand::{prelude::SliceRandom, rngs::StdRng, SeedableRng};
+use std::sync::Arc;
+
+use consensus_config::{AuthorityIndex, Committee};
+
+/// The LeaderSchedule is responsible for producing the leader schedule across
+/// an epoch. For now it is a simple wrapper around committee to provide a leader
+/// for a round deterministically.
+// TODO: complete full leader schedule changes
+#[derive(Clone)]
+pub struct LeaderSchedule {
+    committee: Arc<Committee>,
+}
+
+#[allow(unused)]
+impl LeaderSchedule {
+    pub fn new(committee: Arc<Committee>) -> Self {
+        Self { committee }
+    }
+
+    pub fn elect_leader(&self, round: u32, leader_offset: u32) -> AuthorityIndex {
+        cfg_if::cfg_if! {
+            // TODO: we need to differentiate the leader strategy in tests, so for
+            // some type of testing (ex sim tests) we can use the staked approach.
+            if #[cfg(test)] {
+                AuthorityIndex::new_for_test((round + leader_offset) % self.committee.size() as u32)
+            } else {
+                self.elect_leader_stake_based(round, leader_offset)
+            }
+        }
+    }
+
+    pub fn elect_leader_stake_based(&self, round: u32, offset: u32) -> AuthorityIndex {
+        assert!((offset as usize) < self.committee.size());
+
+        // TODO: this needs to be removed.
+        // if genesis, always return index 0
+        if round == 0 {
+            return AuthorityIndex::new_for_test(0);
```

Comment:
I don't think specializing for round 0 is necessary?

Reply:
Tested it in mysticeti, its safe to remove

Reply:
@arun-koshy how did you test this (because this code won't run in the sim tests or unit tests), in this AWS benchmark tests? I explicitly had to do this work around back then because otherwise we could end up with inconsistent leader election which was leading to a failure on committing. Now I did various changes since then so I am not sure it is indeed needed anymore.

Reply:
Spoke with @akichidis offline, but to test, I removed this check and used stake based leader election in the tests in the mysticeti codebase and everything passed. But to be sure nothing was missed also triggered a deployment without this check and rounds were able to advance.

https://metrics.sui.io/goto/bKyE-m5SR?orgId=1